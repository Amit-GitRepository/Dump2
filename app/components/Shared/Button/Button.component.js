import merge from 'lodash/merge';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React from 'react';
import {Icon, ISText, Touchable} from '../../Shared';
import {View} from 'react-native';
import * as ButtonStyles from './Button.style';

const Button = ({accessibilityLabel, text, type, iconName, style, disabled, iconStyle, textStyle, textType, testID, touchableStyle, onPress}) => {
  const styles = merge({}, ButtonStyles.baseStyles, ButtonStyles[`${type}Styles`]);
  const containerStyle = disabled ? [styles.container, style, styles.disabled, touchableStyle] : [styles.container, style, touchableStyle];
  return (
    <Touchable borderless={true} style={containerStyle} accessibilityLabel={accessibilityLabel} testID={testID} disabled={disabled} onPress={disabled ? noop : onPress} accessible={true}>
      <View style={styles.wrapper}>
        {iconName ? <Icon style={[styles.icon, iconStyle]} name={iconName} /> : null}
        <ISText type={textType} style={[styles.text, textStyle]}>{text}</ISText>
      </View>
    </Touchable>
  );
};

Button.defaultProps = {
  style: {},
  iconStyle: {},
  textStyle: {},
  touchableStyle: {},
  textType: 'BOLD',
  text: '',
  iconName: null,
  onPress: noop,
  type: 'primary'
};

Button.propTypes = {
  accessibilityLabel: PropTypes.string,
  style: PropTypes.object,
  iconStyle: PropTypes.object,
  touchableStyle: PropTypes.object,
  textStyle: PropTypes.object,
  text: PropTypes.string,
  iconName: PropTypes.string,
  type: PropTypes.oneOf(['primary', 'secondary', 'inline']),
  disabled: PropTypes.bool,
  onPress: PropTypes.func,
  textType: PropTypes.oneOf(['REGULAR', 'BOLD', 'LIGHT', 'MEDIUM', 'SEMIBOLD', 'THIN']),
  testID: PropTypes.string
};

export default Button;
