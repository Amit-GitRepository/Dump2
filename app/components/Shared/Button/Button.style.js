import {colors, fonts} from '../../../themes/constants.styles';

export const secondaryStyles = {
  text: {
    color: colors.PRIMARY_TEXT_TAB_LABEL
  },
  icon: {
    color: colors.PRIMARY_ACTIONABLE
  },
  container: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    borderColor: colors.PRIMARY_TEXT_TAB_LABEL,
    borderWidth: 1.5
  }
};

export const inlineStyles = {
  text: {
    color: colors.PRIMARY_ACTIONABLE,
    fontSize: fonts.FONT_SIZE_NORMAL
  },
  container: {
    padding: 5,
    borderWidth: 0,
    backgroundColor: colors.TRANSPARENT
  }
};

export const primaryStyles = {
  container: {
    backgroundColor: colors.PRIMARY_ACTIONABLE
  },
  text: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  }
};

export const baseStyles = {
  container: {
    alignSelf: 'stretch',
    padding: 10,
    borderRadius: 50
  },
  disabled: {
    opacity: 0.7
  },
  wrapper: {
    flexDirection: 'row',
    alignSelf: 'stretch',
    justifyContent: 'center',
    alignItems: 'center'
  },
  icon: {
    paddingHorizontal: 10,
    fontSize: fonts.FONT_SIZE_LARGE,
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  },
  text: {
    fontSize: fonts.FONT_SIZE_NORMAL
  }
};
