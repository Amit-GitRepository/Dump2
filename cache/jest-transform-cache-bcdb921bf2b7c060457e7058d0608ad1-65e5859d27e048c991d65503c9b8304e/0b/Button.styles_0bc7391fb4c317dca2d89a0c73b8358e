Object.defineProperty(exports,"__esModule",{value:true});exports.baseStyles=exports.primaryStyles=exports.inlineStyles=exports.secondaryStyles=undefined;var cov_2kmh9kgj0r=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Button/Button.styles.js',hash='22d23e82d9cfa8634c3ca8f2fd06e03b512b7653',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Button/Button.styles.js',statementMap:{'0':{start:{line:3,column:31},end:{line:15,column:1}},'1':{start:{line:17,column:28},end:{line:27,column:1}},'2':{start:{line:29,column:29},end:{line:36,column:1}},'3':{start:{line:38,column:26},end:{line:61,column:1}}},fnMap:{},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');

var secondaryStyles=exports.secondaryStyles=(cov_2kmh9kgj0r.s[0]++,{
text:{
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL},

icon:{
color:_constants.colors.PRIMARY_ACTIONABLE},

container:{
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
borderColor:_constants.colors.PRIMARY_TEXT_TAB_LABEL,
borderWidth:1.5}});



var inlineStyles=exports.inlineStyles=(cov_2kmh9kgj0r.s[1]++,{
text:{
color:_constants.colors.PRIMARY_ACTIONABLE,
fontSize:_constants.fonts.FONT_SIZE_NORMAL},

container:{
padding:5,
borderWidth:0,
backgroundColor:_constants.colors.TRANSPARENT}});



var primaryStyles=exports.primaryStyles=(cov_2kmh9kgj0r.s[2]++,{
container:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE},

text:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST}});



var baseStyles=exports.baseStyles=(cov_2kmh9kgj0r.s[3]++,{
container:{
alignSelf:'stretch',
padding:10,
borderRadius:50},

disabled:{
opacity:0.7},

wrapper:{
flexDirection:'row',
alignSelf:'stretch',
justifyContent:'center',
alignItems:'center'},

icon:{
paddingHorizontal:10,
fontSize:_constants.fonts.FONT_SIZE_LARGE,
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

text:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL}});