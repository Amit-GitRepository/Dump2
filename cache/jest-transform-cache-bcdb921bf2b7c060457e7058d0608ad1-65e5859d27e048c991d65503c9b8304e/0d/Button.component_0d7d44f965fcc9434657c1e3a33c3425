Object.defineProperty(exports,"__esModule",{value:true});var cov_1th59psrt6=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Button/Button.component.js',hash='f5d0d00c0176fa5f05aa2bc1cf65f3772a7e1e98',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Button/Button.component.js',statementMap:{'0':{start:{line:11,column:15},end:{line:24,column:1}},'1':{start:{line:12,column:17},end:{line:12,column:82}},'2':{start:{line:13,column:25},end:{line:13,column:106}},'3':{start:{line:14,column:2},end:{line:23,column:4}},'4':{start:{line:26,column:0},end:{line:35,column:2}},'5':{start:{line:37,column:0},end:{line:47,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:11,column:15},end:{line:11,column:16}},loc:{start:{line:11,column:101},end:{line:24,column:1}},line:11}},branchMap:{'0':{loc:{start:{line:13,column:25},end:{line:13,column:106}},type:'cond-expr',locations:[{start:{line:13,column:36},end:{line:13,column:78}},{start:{line:13,column:81},end:{line:13,column:106}}],line:13},'1':{loc:{start:{line:15,column:44},end:{line:15,column:69}},type:'cond-expr',locations:[{start:{line:15,column:55},end:{line:15,column:59}},{start:{line:15,column:62},end:{line:15,column:69}}],line:15},'2':{loc:{start:{line:18,column:11},end:{line:18,column:86}},type:'cond-expr',locations:[{start:{line:18,column:22},end:{line:18,column:79}},{start:{line:18,column:82},end:{line:18,column:86}}],line:18}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0},f:{'0':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _Icon=require('../Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _ISText=require('../ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _merge=require('lodash/merge');var _merge2=_interopRequireDefault(_merge);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _Touchable=require('../Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _reactNative=require('react-native');
var _Button=require('./Button.styles');var ButtonStyles=_interopRequireWildcard(_Button);function _interopRequireWildcard(obj){if(obj&&obj.__esModule){return obj;}else{var newObj={};if(obj!=null){for(var key in obj){if(Object.prototype.hasOwnProperty.call(obj,key))newObj[key]=obj[key];}}newObj.default=obj;return newObj;}}function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}cov_1th59psrt6.s[0]++;

var Button=function Button(_ref){var text=_ref.text,type=_ref.type,iconName=_ref.iconName,style=_ref.style,disabled=_ref.disabled,iconStyle=_ref.iconStyle,textStyle=_ref.textStyle,textType=_ref.textType,onPress=_ref.onPress;cov_1th59psrt6.f[0]++;
var styles=(cov_1th59psrt6.s[1]++,(0,_merge2.default)({},ButtonStyles.baseStyles,ButtonStyles[type+'Styles']));
var touchableStyle=(cov_1th59psrt6.s[2]++,disabled?(cov_1th59psrt6.b[0][0]++,[styles.container,style,styles.disabled]):(cov_1th59psrt6.b[0][1]++,[styles.container,style]));cov_1th59psrt6.s[3]++;
return(
_react2.default.createElement(_Touchable2.default,{disabled:disabled,onPress:disabled?(cov_1th59psrt6.b[1][0]++,_noop2.default):(cov_1th59psrt6.b[1][1]++,onPress)},
_react2.default.createElement(_reactNative.View,{style:touchableStyle},
_react2.default.createElement(_reactNative.View,{style:styles.wrapper},
iconName?(cov_1th59psrt6.b[2][0]++,_react2.default.createElement(_Icon2.default,{style:[styles.icon,iconStyle],name:iconName})):(cov_1th59psrt6.b[2][1]++,null),
_react2.default.createElement(_ISText2.default,{type:textType,style:[styles.text,textStyle]},text)))));




};cov_1th59psrt6.s[4]++;

Button.defaultProps={
style:{},
iconStyle:{},
textStyle:{},
textType:'BOLD',
text:'',
iconName:null,
onPress:_noop2.default,
type:'primary'};cov_1th59psrt6.s[5]++;


Button.propTypes={
style:_propTypes2.default.object,
iconStyle:_propTypes2.default.object,
textStyle:_propTypes2.default.object,
text:_propTypes2.default.string,
iconName:_propTypes2.default.string,
type:_propTypes2.default.oneOf(['primary','secondary','inline']),
disabled:_propTypes2.default.bool,
onPress:_propTypes2.default.func,
textType:_propTypes2.default.oneOf(['REGULAR','BOLD','LIGHT','MEDIUM','SEMIBOLD','THIN'])};exports.default=


Button;