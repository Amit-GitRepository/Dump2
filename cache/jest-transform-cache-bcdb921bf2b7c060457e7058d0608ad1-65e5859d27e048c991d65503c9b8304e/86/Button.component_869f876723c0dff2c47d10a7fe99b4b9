Object.defineProperty(exports,"__esModule",{value:true});var cov_1th59psrt6=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Button/Button.component.js',hash='0505c1d4c68bb7bf629b50141678a2aa38a8b9e1',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Button/Button.component.js',statementMap:{'0':{start:{line:9,column:15},end:{line:22,column:1}},'1':{start:{line:10,column:17},end:{line:10,column:82}},'2':{start:{line:11,column:25},end:{line:11,column:106}},'3':{start:{line:12,column:2},end:{line:21,column:4}},'4':{start:{line:24,column:0},end:{line:34,column:2}},'5':{start:{line:36,column:0},end:{line:49,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:9,column:15},end:{line:9,column:16}},loc:{start:{line:9,column:145},end:{line:22,column:1}},line:9}},branchMap:{'0':{loc:{start:{line:11,column:25},end:{line:11,column:106}},type:'cond-expr',locations:[{start:{line:11,column:36},end:{line:11,column:78}},{start:{line:11,column:81},end:{line:11,column:106}}],line:11},'1':{loc:{start:{line:13,column:123},end:{line:13,column:148}},type:'cond-expr',locations:[{start:{line:13,column:134},end:{line:13,column:138}},{start:{line:13,column:141},end:{line:13,column:148}}],line:13},'2':{loc:{start:{line:16,column:11},end:{line:16,column:86}},type:'cond-expr',locations:[{start:{line:16,column:22},end:{line:16,column:79}},{start:{line:16,column:82},end:{line:16,column:86}}],line:16}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0},f:{'0':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _merge=require('lodash/merge');var _merge2=_interopRequireDefault(_merge);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _Shared=require('../../Shared');
var _reactNative=require('react-native');
var _Button=require('./Button.style');var ButtonStyles=_interopRequireWildcard(_Button);function _interopRequireWildcard(obj){if(obj&&obj.__esModule){return obj;}else{var newObj={};if(obj!=null){for(var key in obj){if(Object.prototype.hasOwnProperty.call(obj,key))newObj[key]=obj[key];}}newObj.default=obj;return newObj;}}function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}cov_1th59psrt6.s[0]++;

var Button=function Button(_ref){var accessibilityLabel=_ref.accessibilityLabel,text=_ref.text,type=_ref.type,iconName=_ref.iconName,style=_ref.style,disabled=_ref.disabled,iconStyle=_ref.iconStyle,textStyle=_ref.textStyle,textType=_ref.textType,testID=_ref.testID,onPress=_ref.onPress,touchableStyle=_ref.touchableStyle;cov_1th59psrt6.f[0]++;
var styles=(cov_1th59psrt6.s[1]++,(0,_merge2.default)({},ButtonStyles.baseStyles,ButtonStyles[type+'Styles']));
var containerStyle=(cov_1th59psrt6.s[2]++,disabled?(cov_1th59psrt6.b[0][0]++,[styles.container,style,styles.disabled]):(cov_1th59psrt6.b[0][1]++,[styles.container,style]));cov_1th59psrt6.s[3]++;
return(
_react2.default.createElement(_Shared.Touchable,{style:touchableStyle,accessibilityLabel:accessibilityLabel,testID:testID,disabled:disabled,onPress:disabled?(cov_1th59psrt6.b[1][0]++,_noop2.default):(cov_1th59psrt6.b[1][1]++,onPress),accessible:true},
_react2.default.createElement(_reactNative.View,{style:containerStyle},
_react2.default.createElement(_reactNative.View,{style:styles.wrapper},
iconName?(cov_1th59psrt6.b[2][0]++,_react2.default.createElement(_Shared.Icon,{style:[styles.icon,iconStyle],name:iconName})):(cov_1th59psrt6.b[2][1]++,null),
_react2.default.createElement(_Shared.ISText,{type:textType,style:[styles.text,textStyle]},text)))));




};cov_1th59psrt6.s[4]++;

Button.defaultProps={
style:{},
iconStyle:{},
textStyle:{},
textType:'BOLD',
text:'',
touchableStyle:{},
iconName:null,
onPress:_noop2.default,
type:'primary'};cov_1th59psrt6.s[5]++;


Button.propTypes={
accessibilityLabel:_propTypes2.default.string,
style:_propTypes2.default.object,
iconStyle:_propTypes2.default.object,
textStyle:_propTypes2.default.object,
touchableStyle:_propTypes2.default.object,
text:_propTypes2.default.string,
iconName:_propTypes2.default.string,
type:_propTypes2.default.oneOf(['primary','secondary','inline']),
disabled:_propTypes2.default.bool,
onPress:_propTypes2.default.func,
textType:_propTypes2.default.oneOf(['REGULAR','BOLD','LIGHT','MEDIUM','SEMIBOLD','THIN']),
testID:_propTypes2.default.string};exports.default=


Button;