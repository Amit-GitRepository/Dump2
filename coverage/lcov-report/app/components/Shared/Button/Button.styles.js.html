<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Shared/Button/Button.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">app/components/Shared/Button</a> Button.styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {colors, fonts} from '../../../themes/constants.styles';
&nbsp;
export const secondaryStyles = {
  text: {
    color: colors.PRIMARY_TEXT_TAB_LABEL
  },
  icon: {
    color: colors.PRIMARY_ACTIONABLE
  },
  container: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    borderColor: colors.PRIMARY_TEXT_TAB_LABEL,
    borderWidth: 1.5
  }
};
&nbsp;
export const inlineStyles = {
  text: {
    color: colors.PRIMARY_ACTIONABLE,
    fontSize: fonts.FONT_SIZE_NORMAL
  },
  container: {
    padding: 5,
    borderWidth: 0,
    backgroundColor: colors.TRANSPARENT
  }
};
&nbsp;
export const primaryStyles = {
  container: {
    backgroundColor: colors.PRIMARY_ACTIONABLE
  },
  text: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  }
};
&nbsp;
export const baseStyles = {
  container: {
    alignSelf: 'stretch',
    padding: 10,
    borderRadius: 50
  },
  disabled: {
    opacity: 0.7
  },
  wrapper: {
    flexDirection: 'row',
    alignSelf: 'stretch',
    justifyContent: 'center',
    alignItems: 'center'
  },
  icon: {
    paddingHorizontal: 10,
    fontSize: fonts.FONT_SIZE_LARGE,
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  },
  text: {
    fontSize: fonts.FONT_SIZE_NORMAL
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 31 2017 11:24:42 GMT+0530 (IST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
