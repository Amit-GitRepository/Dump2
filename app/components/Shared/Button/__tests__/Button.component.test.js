import Button from '../Button.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('Button component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button/>);
    const component = renderer.create(<Button />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
  it('renders correctly with icon', () => {
    const wrapper = shallow(<Button iconName='trueid'/>);
    expect(wrapper).toBeDefined();
  });
});
