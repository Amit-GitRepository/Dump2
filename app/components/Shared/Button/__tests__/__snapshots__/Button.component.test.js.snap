// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#F93E60",
      "borderRadius": 50,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#303C46",
            },
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
              },
              Object {},
            ],
            Object {
              "fontFamily": "SukhumvitSet-Bold",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
